src/DoctrineORMModule/Service/DBALConnectionFactory.php

```diff
@@ -59,6 +59,10 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
         foreach ($options->getDoctrineCommentedTypes() as $type) {
             $connection->getDatabasePlatform()->markDoctrineTypeCommented(Type::getType($type));
         }
+        
+        if ($options->useSavepoints()) {
+            $connection->setNestTransactionsWithSavepoints(true);
```

Comment:
Because savepoints is defaulted to false you can drop the if statement here and just use
```
$connection->setNestTransactionsWithSavepoints($options->useSavepoints());
```

Reply:
I can't do this, because the `setNestTransactionsWithSavepoints()` method in the connection class always checks if the platform supports it and throws an exception if it doesn't (even when passing `false`).